src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const resultObject: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const objKey = item[key] as string;
+
+    if (!resultObject[objKey]) {
+      resultObject[objKey] = [item];
+    } else {
+      resultObject[objKey].push(item);
+    }
```

Comment:
Prefer don't use else, you can create an empty array if doesn't exist and in both cases just push an item.